platform-server-kernel/src/main/java/uno/cod/platform/server/core/dto/task/TaskShowDto.java

```diff
@@ -4,21 +4,27 @@
 import uno.cod.platform.server.core.domain.CodingSkill;
 import uno.cod.platform.server.core.domain.Task;
 import uno.cod.platform.server.core.dto.assignment.AssignmentShowDto;
+import uno.cod.platform.server.core.dto.language.LanguageShowDto;
 import uno.cod.platform.server.core.dto.template.TemplateShowDto;
 
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
+import java.util.stream.Collectors;
 
 public class TaskShowDto extends AssignmentShowDto{
     public TaskShowDto(Task task){
         BeanUtils.copyProperties(task, this);
         this.normalRunAvailable = task.getRunner() != null;
+        if(task.getLanguages()!=null){
+            this.languages = task.getLanguages().stream().map(LanguageShowDto::new).collect(Collectors.toList());
```

Comment:
I was wondering whether this would preserve the order of the original list. Turns out yes, in case you are interested check [this](http://stackoverflow.com/questions/30258566).